packages/server/billing/helpers/adjustUserCount.ts

```diff
@@ -84,10 +81,7 @@ const addUser = async (orgIds: string[], userId: string) => {
       new Date()
     return new OrganizationUser({orgId, userId, newUserUntil, tier: organization.tier})
   })
-  await r
-    .table('OrganizationUser')
-    .insert(docs)
-    .run()
+  await r.table('OrganizationUser').insert(docs, {conflict: 'replace'}).run()
```

Comment:
Makes sense, thanks. I just wanted to give you an update. I've added checks to see if the user exists in the org to prevent replication and it kinda works but not perfectly. 

When replicating the `AcceptTeamInvitationMutation` 5x, sometimes the 2nd or 3rd (or 5th) function invocation fires and reads the `OrganizationUser` table before the first `addUser` insert has completed. `userInOrg` is therefore false and it replicates the same user.

I've pushed an update that checks if the user already exists in the org just before the insert in `adjustUserCounts`. This helps and moving the `userInOrg` check below the `Promise.all` also gives it more time and reduces replication. 

But it's not a perfect solution. Occasionally, the user is still replicated. Strangely, it seems to happen with more frequency if I remove the user from the org & re-add them.

A few potential solutions I'm thinking of:

1. I can't see any code that checks if a user joins/leaves/joins/leaves. If that's the case, perhaps we could make `removedAt` and `joinedAt` and array instead and keep track of joining & leaving that way? Then we could use `userId::orgId` as the id
2. Try to limit the number of times the client mutation can be called
3. Adding timeouts or a function that removes duplicates at the end of `acceptTeamInvitation`. This seems very hacky

Do you have any other ideas I could explore?

Reply:
can you point to the race condition in the code? i'd think the easiest solution would be to make sure that the user has been inserted before continuing on

Reply:
Sure thing, the new `OrganizationUser` is being inserted [here](https://github.com/ParabolInc/parabol/blob/fix-replicated-user/packages/server/billing/helpers/adjustUserCount.ts#L96).

Do you mean check whether the user has not been inserted before inserting? If so, I'm doing this [here](https://github.com/ParabolInc/parabol/blob/fix-replicated-user/packages/server/billing/helpers/adjustUserCount.ts#L85) and [here](https://github.com/ParabolInc/parabol/blob/fix-replicated-user/packages/server/safeMutations/acceptTeamInvitation.ts#L109). 

The inserting is happening successfully so I'm not sure we'd need to check if the user has been inserted before continuing. The problem is the user is being inserted when it already exists. 

I also tried adding a `branch` statement to check if the user exists at the final moment and only insert if it doesn't but no dice. It seems that the 2nd/3rd function invocation reads from `OrganizationUser` before the 1st insert has happened so the checks return false